routers/sbp_register.py

```diff
@@ -24,6 +24,16 @@
     route_class=RegistrationRoute
 )
 
+def _validate_email_domain_with_list(email: str, allowed_domains: list[str]) -> bool:
+    email_lower = email.lower()
+    return any(email_lower.endswith(domain) for domain in allowed_domains)
+
+
+def validate_sbp_email_domain(email: str) -> bool:
```

Comment:
make the settings an argument to this function, and pass it through from the endpoint that calls it.

`get_settings()` is potentially slow, so it's best to let FastAPI handle it via its dependencies (which ensures it gets called at most once per route)